### ISSUE TITLE
[Enhancement] Cache VideoJS dependencies for docker to reduce build times

### ISSUE THREAD

**Issue by syeopite** [iv-org/invidious#4878]
**Is your enhancement request related to a problem? Please describe.**

The `Checking player dependencies` part of compiling Invidious takes a huge amount of time on docker builds. 

**Describe the solution you'd like**

The VideoJS dependencies should be cached to mitigate this problem. Although I'm not sure if the lengthy part comes from compiling the VideoJS dependency fetch script itself or from the actual fetching of the dependencies themselves from NPM.

This should be investigated

**Describe alternatives you've considered**

**Additional context**

**Comment by SamantazFox** [comment]
Yes, I was thinking about making that its own step in the makefile. That would also allow creating release tarballs (requested in #4672)

**Comment by syeopite** [comment]
Something like this? https://github.com/iv-org/invidious/pull/4300
